x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/Phase.java

```diff
@@ -40,7 +44,19 @@
                     .collect(Collectors.toMap(LifecycleAction::getWriteableName, Function.identity()))));
     static {
         PARSER.declareField(ConstructingObjectParser.optionalConstructorArg(),
-                (p, c) -> TimeValue.parseTimeValue(p.text(), MIN_AGE.getPreferredName()), MIN_AGE, ValueType.VALUE);
+            (ContextParser<String, Object>) (p, c) -> {
+                // In earlier versions it was possible to create a Phase with a negative `min_age` which would then cause errors
+                // when the phase is read from the cluster state during startup (even before negative timevalues were strictly
+                // disallowed) so this is a hack to treat negative `min_age`s as 0 to prevent those errors.
+                // They will be saved as `0` so this hack can be removed once we no longer have to read cluster states from 7.x.
```

Comment:
Can you add an assert Version.CURRENT.major < 9 so that we remove this when we bump master to 9.0.0?

Reply:
Done.